lib/persisted-model.js

```diff
@@ -993,15 +993,8 @@ module.exports = function(registry) {
         {verb: 'post', path: '/change-stream'},
         {verb: 'get', path: '/change-stream'},
       ],
-      accepts: {
-        arg: 'options',
-        type: 'object',
-      },
-      returns: {
-        arg: 'changes',
-        type: 'ReadableStream',
-        json: true,
-      },
+      accepts: {arg: 'options', type: 'object', http: 'optionsFromRequest'},
```

Comment:
I did not realise that `createChangeStream` was already exposing the `options` argument. The change proposed in this pull request can break existing applications, because right now it's possible to specify options via query string:

```
GET /api/mymodels/change-stream?options[flag]=value
-> options = {flag: 'value'}
```

We need to preserve backwards compatibility. The solution is to add a feature flag (a model-level setting) that will control whether the `options` argument should be provided by the user (current state) or constructed via `optionsFromRequest` (the new version).

Here is a quick mock-up:

```js
var streamOptionsArg = {arg: 'options', type: 'object'};
// TODO: remove this flag in 4.0
if (PersistedModel.settings.injectChangeStreamOptions) {
  streamOptionsArg.http = 'optionsFromRequest';
}

setRemoting(PersistedModel, 'createChangeStream', {
  // ...,
  accepts: streamOptionsArg,
  returns: // ...
});
```